src/DynamicNLPModels.jl

```diff
@@ -318,7 +364,125 @@ function LQDynamicModel(
     )
 end
 
+function _build_condensed_blocks(
+    s0, Q, R, A, B, N;
+    Qf = Q)
+  
+    ns = size(Q,1)
+    nu = size(R,1)
+  
+    # Define block matrices
+    block_B = zeros(ns*(N+1), nu*(N))
+    block_A = zeros(ns*(N+1), ns)
+    block_Q = SparseArrays.sparse([],[], Float64[], ns*(N+1), ns*(N+1))
+    block_R = SparseArrays.sparse([],[], Float64[], nu*(N), nu*(N))
+  
+    block_A[1:ns, 1:ns] .= Matrix(LinearAlgebra.I, ns, ns)
+  
+    # Define matrices for mul!
+    A_klast  = copy(A)
+    A_k      = similar(A)
```

Comment:
`similar` creates uninitialized array. As a rule of thumb, I strongly advise to fill them with zero to avoid non-deterministic behavior:
```julia
A_k = similar(A) ; fill!(A, 0.0)
```